### ISSUE TITLE
Persist Scratchpad

### ISSUE THREAD

**Issue by hediet** [hediet/vscode-drawio#171]
Currently, the scratchpad is lost when closing and reopening a diagram.

**Comment by hediet** [comment]
Draw.io uses indexeddb to persist the scratchpad.
Probably this can be used to serialize/deserialize it: https://www.npmjs.com/package/indexeddb-export-import

**Comment by bpasero** [comment]
I was testing this today because I was curious how well this extension is supporting what we call "hot exit" in VSCode. As a user when I open editors (text or custom editors like draw.io) and make changes, I would expect that everything restores fine even when I am not saving. 

If the custom editor is backed by text documents, there is actually nothing you would need to be doing because we take care of backups then (`CustomTextEditorProvider`). I assume the extension is registering a `CustomEditorProvider`?

**Comment by hediet** [comment]
Hi ;) Yes, it is registering a custom editor provider.

The problem is that the scratchpad is not persisted in the diagram (which kind of makes sense, as the scratchpad is like snippets and should be a user setting). Instead they use indexeddb to persist the data in the local storage of the window. This storage is not persisted by VS Code (I would not expect this anyways).

The solution would be to watch the indexeddb of the web view and send the content over to the VS Code extension everytime it changes. The extension would persist it in the user/workspace settings.
Maybe draw.io even has an abstraction over indexeddb that allows for direct serialization, but I doubt that.

**Comment by bpasero** [comment]
Yeah makes sense, it sounds like having a way for a user of the draw.io library to fully provide the data persistence would help here and would be required.

**Comment by tonyawad88** [comment]
In case someone is looking to persist the scratchpad for the time being, you can download draw.io desktop and toggle the attach settings in vscode to "off" in order to use the desktop app instance within instead of the bundled one. 
![image](https://user-images.githubusercontent.com/3638322/167277405-7e3dcd09-ec77-4a74-b952-3caba3d258a7.png)

**Comment by masongyli** [comment]
@tonyawad88 Thank you, it works. But after toggling this on, I cannot simply "Save" when editing a graph. If I want to save the graph, I need to "Save & Exit".

**Comment by AnatolyIliev** [comment]
@tonyawad88 
> use the desktop app instance within instead of the bundled one.

What do you mean? Doesn't that make it use the online version instead?

 ![text](https://i.imgur.com/q3XFLDP.png)

Whatever way you got this to work, I hope it makes custom shape libraries persistent as well.

**Comment by AnatolyIliev** [comment]
@tonyawad88 @masongyli @debojyoti
> use the desktop app instance within instead of the bundled one.

What do you mean? Doesn't that make it use the online version instead?

 ![text](https://i.imgur.com/q3XFLDP.png)

Whatever way you got this to work, I hope it makes custom shape libraries persistent as well.

**Comment by AnatolyIliev** [comment]
pls respond ;_;

**Comment by iFUCKINGHATEcomputers** [comment]
@tonyawad88 @masongyli @debojyoti @heheym @hediet
guise cmon, how do you use another local instance with this?

**Comment by AnatolyIliev** [comment]
@tonyawad88 @masongyli @debojyoti @heheym @hediet
guise cmon, how do you use another local instance with this?
